lesson17/src/js/main.js

```diff
@@ -0,0 +1,212 @@
+  separation.textContent = "/";
+  separation.classList.add("separation");
+  fragment.appendChild(separation);
+
+  const total = document.createElement("span");
+  total.id = 'js-total';
+  total.textContent = item.length;
+  fragment.appendChild(total);
+
+  const textArea = document.querySelector(".mainvisual__textarea");
+  textArea.appendChild(pagenation).appendChild(fragment);
+}
+
+function manipulatePagination(target) {
+  const pagenationCurrent = document.getElementById("js-current");
+  const pagenationTotal = Number(document.getElementById("js-total").textContent);
+  let currentNum = Number(pagenationCurrent.textContent);
+
+  const buttonPrev = target.getAttribute("aria-label") === 'previous';
+  const buttonNext = target.getAttribute("aria-label") === 'next';
+
+  if (currentNum < pagenationTotal && buttonNext) {
+    pagenationCurrent.textContent = currentNum +=1;
+  }else if(currentNum <= pagenationTotal && buttonPrev){
+    pagenationCurrent.textContent = currentNum -= 1;
+  }
+}
+
+function slideImageNext(e) {
+  const slideImages = [...document.getElementsByClassName("slider__item")];
+
+  const activeSlide = document.querySelector(".is-active");
+  const nextSlide = activeSlide.nextElementSibling;
+  const nextSlideIndex = Number(nextSlide.getAttribute("data-slide-index"));
+  if (nextSlideIndex === slideImages.length) {
+    nextSlide.classList.add("is-active");
+    activeSlide.classList.remove("is-active");
+    e.disabled = true;
+  } else {
+    e.previousElementSibling.disabled = false;
+    nextSlide.classList.add("is-active");
+    activeSlide.classList.remove("is-active");
+  }
+}
+
+function slideImagePrev(e) {
+  const activeSlide = document.querySelector(".is-active");
+  const prevSlide = activeSlide.previousElementSibling;
+  const prevSlideIndex = Number(prevSlide.getAttribute("data-slide-index"));
+
+  if (prevSlideIndex === 1) {
+    prevSlide.classList.add("is-active");
+    activeSlide.classList.remove("is-active");
+    e.disabled = true;
+  } else {
+    e.nextElementSibling.disabled = false;
+    prevSlide.classList.add("is-active");
+    activeSlide.classList.remove("is-active");
+  }
+}
+
+function callImageData() {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(fetchImageData("https://api.json-generator.com/templates/9tm12BO1y5Xx/data?access_token=b0154huvd1stffra1six9olbgg34r4zofcqgwzfl&delay=1000"));
+    }, 3000);
+  });
+}
+
+async function fetchImageData(URL) {
+  const response = await fetch(URL);
+  if (response.ok) {
+    const json = await response.json();
+    return json;
+  } else {
+    console.error(`${response.status}:${response.statusText}`);
+    displayErrorMassage("サーバの通信に失敗しました");
+  }
+}
+
+async function callSlideContents() {
+  showLoadingImage();
+  try {
+    const json = await callImageData();
+    const slideImageData = json.slide;
+
+    renderSlideImage(slideImageData);
+    renderPagenation(slideImageData);
```

Comment:
In renderPagenation(slideImageData), I thought it would be a good idea to pass the length instead of all the data at this point, since only the length is used by the executor.
```suggestion
    renderPagenation(slideImageData.length);
```

Reply:
Thank you for your review!
Certainly it is smarter to pass only the necessary data here!

[0aeb610](https://github.com/nyattsu72/hanson-frontend-cource/pull/25/commits/0aeb610893f61a84249e0b9e560ffe255d4fa3ab)